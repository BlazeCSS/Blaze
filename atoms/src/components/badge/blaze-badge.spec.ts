import { flush, render } from '@stencil/core/testing';
import { Badge } from './blaze-badge';

describe('Badge', () => {
  it('should build', () => {
    expect(new Badge()).toBeTruthy();
  });

  let element;

  const snapIt = (name, html) => {
    it(name, async () => {
      element = await render({
        components: [Badge],
        html
      });
      await flush(element);

      expect(element).toMatchSnapshot();
    });
  };

  snapIt(
    'renders correctly with no props',
    '<blaze-badge>default</blaze-badge>'
  );

  snapIt(
    'appends the color modifier class',
    '<blaze-badge type="brand">brand</blaze-badge>'
  );

  snapIt(
    'handles boolean rounded attribute',
    '<blaze-badge rounded>default</blaze-badge>'
  );

  snapIt(
    'handles both boolean rounded attribute and type',
    '<blaze-badge rounded type="brand">brand</blaze-badge>'
  );

  snapIt(
    'handles boolean ghost and boolean rounded',
    '<blaze-badge ghost rounded>default</blaze-badge>'
  );

  snapIt(
    'supports all the modifiers',
    '<blaze-badge ghost rounded type="brand">brand</blaze-badge>'
  );
});
