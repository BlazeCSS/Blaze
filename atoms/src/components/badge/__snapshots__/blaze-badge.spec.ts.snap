// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge appends the color modifier class 1`] = `
<blaze-badge
  class="hydrated"
  type="brand"
>
  <span
    class="c-text c-badge c-badge--brand"
  >
    <!---->
    brand
  </span>
</blaze-badge>
`;

exports[`Badge handles boolean ghost and boolean rounded 1`] = `
<blaze-badge
  class="hydrated"
  ghost=""
  rounded=""
>
  <span
    class="c-text c-badge c-badge--rounded c-badge--ghost"
  >
    <!---->
    default
  </span>
</blaze-badge>
`;

exports[`Badge handles boolean rounded attribute 1`] = `
<blaze-badge
  class="hydrated"
  rounded=""
>
  <span
    class="c-text c-badge c-badge--rounded"
  >
    <!---->
    default
  </span>
</blaze-badge>
`;

exports[`Badge handles both boolean rounded attribute and type 1`] = `
<blaze-badge
  class="hydrated"
  rounded=""
  type="brand"
>
  <span
    class="c-text c-badge c-badge--brand c-badge--rounded"
  >
    <!---->
    brand
  </span>
</blaze-badge>
`;

exports[`Badge renders correctly with no props 1`] = `
<blaze-badge
  class="hydrated"
>
  <span
    class="c-text c-badge"
  >
    <!---->
    default
  </span>
</blaze-badge>
`;

exports[`Badge supports all the modifiers 1`] = `
<blaze-badge
  class="hydrated"
  ghost=""
  rounded=""
  type="brand"
>
  <span
    class="c-text c-badge c-badge--brand c-badge--rounded c-badge--ghost"
  >
    <!---->
    brand
  </span>
</blaze-badge>
`;
